lib/internal/url.js

```diff
@@ -1406,12 +1410,26 @@ function getPathFromURL(path) {
 // as this is the only character that won't be percent encoded by
 // default URL percent encoding when pathname is set.
 const percentRegEx = /%/g;
-function getURLFromFilePath(filepath) {
-  const tmp = new URL('file://');
-  if (filepath.includes('%'))
-    filepath = filepath.replace(percentRegEx, '%25');
-  tmp.pathname = filepath;
-  return tmp;
+function pathToFileURL(filepath) {
+  let resolved = path.resolve(filepath);
+  // path.resolve strips trailing slashes so we must add them back
+  const filePathLast = filepath.charCodeAt(filepath.length - 1);
+  if ((filePathLast === CHAR_FORWARD_SLASH ||
+       filePathLast === CHAR_BACKWARD_SLASH) &&
+      resolved[resolved.length - 1] !== path.sep)
```

Comment:
Usually forward/back slash code paths have a platform.win32 check to do only forward slash for posix and an additional back slash check for win32.

Reply:
Cool! In posix the `\\` is seen as a legit character in a name.
```
touch foo\\
```
then
```
fs.realpathSync("foo\\")
// '/Users/jdalton/projects/demo/foo\\'
```
What happens in the conversion to URL in that case?

Reply:
guh, probably should be encoded

Reply:
No, the file URL conversion does not support `\\` characters in posix path. `new URL('file:' + '/path/to/x\\')` -> `file:///path/to/x/`.

Reply:
(This is by the WhatWG spec)

Reply:
Thats parsing a URL string, for the path to URL conversion for the path `/x\\y`, it should become `file:///x%5Cy` shouldn't it?

Reply:
@bmeck yes, we could branch this handling for the Windows case. That would be another feature of this conversion that wouldn't apply for naive conversion.

Reply:
Ok I've added the encoding in posix.